moveit_simple/include/moveit_simple/online_robot.h

```diff
@@ -72,8 +72,53 @@ class OnlineRobot : public Robot
    * @throws <moveit_simple::CollisionDetected> (One of interpolated point is
    * in Collision with itself or environment)
    */
-  void execute(std::vector<moveit_simple::JointTrajectoryPoint> &goal,
-    bool collision_check = false);
+  void execute(std::vector<moveit_simple::JointTrajectoryPoint> &goal, bool collision_check = false);
+
+  /**
+   * @brief Set the goal to execute a given trajectory
+   * @param traj_name - name of trajectory to be executed (must be filled with
+   * prior calls to "addTrajPoint".
+   * @param collision_check - bool to turn check for collision on\off
+   */
+  bool setExecuteGoal(const std::string &traj_name, bool collision_check = false);
+
+  /**
+   * @brief Set the goal to execute a given trajectory
+   * @param traj_name - name of trajectory to be executed (must be filled with
+   * prior calls to "addTrajPoint".
+   * @param collision_check - bool to turn check for collision on\off
+   */
+  bool setExecuteGoal(const std::vector<moveit_simple::JointTrajectoryPoint> &goal_points, bool collision_check = false);
+
+  /**
+   * @brief Executes the given planned trajectory
+   */
+  void startExecution();
+
+  /**
+   * @brief Stops executing the given planned trajectory
+   */
+  void stopExecution();
+
+  /**
+   * @brief Checks if the given planned trajectory is in pending / active state
+   */
+  bool isExecuting();
+
+  /**
+   * @brief Checks if the given planned trajectory goal is in collision
+   */
+  bool checkGoalInCollision();
+
+  /**
+   * @brief Checks if the given planned trajectory goal is stopped
+   */
+  bool checkExecutionStopped();
```

Comment:
Does this method check if the execution is DONE or STOPPED (or both)?

Reply:
It checks for the following:

```
action_state == actionlib::SimpleClientGoalState::RECALLED 
action_state == actionlib::SimpleClientGoalState::PREEMPTED
action_state == actionlib::SimpleClientGoalState::ABORTED
```

Reply:
Ok...should this be renamed `is...`?